### PR TITLE
Decoupling encryption key from passcode

### PR DESCRIPTION
**Pull request by wmathurin**
Basically with this PR, we are:
- getting rid of SFKeyStoreKeyType (all keys are generated keys now),
- SFKeyStoreManager only works with a SFGeneratedKeyStore,
- when an app is first unlocked, we check to see if there are any keys in the passcode key store and migrate them to the generated key store.

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCoreTests/SFSecurityTests.m [hunk 1]
```diff
@@ -79,90 +79,65 @@ - (void)testKeyStoreThreadSafety
...
-    // make sure generated-specific key is still in the generated keystore
-    [self assertKeyForDictionary:mgr.generatedKeyStore.keyStoreDictionary
-                       withLabel:[mgr.generatedKeyStore keyLabelForString:@"origKey"]
-                      hasKeyType:SFKeyStoreKeyTypeGenerated
-                hasEncryptionKey:origEncKey];
-}
+#pragma mark - Upgrade tests
 
-- (void)testPasscodeToNoPasscode
+- (void)testUpgradeTo60
```
Comment (wmathurin) [inline libs/SalesforceSDKCore/SalesforceSDKCoreTests/SFSecurityTests.m]: The upgrade test

Comment (wmathurin) [reply libs/SalesforceSDKCore/SalesforceSDKCoreTests/SFSecurityTests.m]: NB: I also tested manually (using SmartSyncExplorerReactNative on 5.3, then changing package.json to point to my fork/branch).

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFKeyStoreManager.m [hunk 2]
```diff
@@ -314,44 +185,14 @@ - (void)observeValueForKeyPath:(NSString *)keyPath ofObject:(id)object change:(N
...
-                // migration should be necessary.
-                [SFSDKCoreLogger i:[self class] format:@"Passcode verified.  Making passcode key store available."];
-                self.passcodeKeyStore.keyStoreKey.encryptionKey.key = [[self class] keyStringToData:newKey];
+
+        if ([oldKey length] == 0 && [newKey length] > 0) {
+            // We just got the passcode, migrate keys (if any)
+            SFPasscodeKeyStore *passcodeKeyStore = [[SFPasscodeKeyStore alloc] init];
+            passcodeKeyStore.keyStoreKey.encryptionKey.key = [[self class] keyStringToData:newKey];
+            if (passcodeKeyStore.keyStoreKey != nil && passcodeKeyStore.keyStoreDictionary.count > 0) {
+                [self migratePasscodeToGenerated:passcodeKeyStore];
```
Comment (wmathurin) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFKeyStoreManager.m]: That's where we handle upgrade

**Comment by huminzhi** [comment]
## Danger has errored
[!] Invalid `DangerFile` file: `minimum_coverage_percentage` value must be a Float! Found Integer instead.
```
 #  from DangerFile:5
 #  -------------------------------------------
 #  
 >  xcov.report(
 #     scheme: 'UnitTests',
 #  -------------------------------------------
```

<p align="right" data-meta="generated_by_CI-result">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**